packages/app/src/providers/DataProvider.tsx

```diff
@@ -102,6 +104,13 @@ function DataProvider(props: PropsWithChildren<Props>) {
     });
   }, [api]);
 
+  const { getSearchablePaths } = api;
+  const searchablePaths = getSearchablePaths
+    ? createFetchStore(async (path: string) =>
+        getSearchablePaths.bind(api)(path)
+      )
+    : undefined;
```

Comment:
Main discussion point here: the exposed `getSearchablePaths` could be simply `api.getSearchablePaths?.bind(api)` (as `getExportURL`).

**However** I would like the result to be cached so I used a `fetchStore` but this seems a bit overengineered to store a list of paths. Moreover, I don't expect to call `getSearchablePaths` with another `path` in arg than `/`.

Reply:
Yeah, I wouldn't bother with the fetch store, especially since we've been talking about refactoring all this fetching business.